kubernetes/mlflow-tracking-operator-master/README.md

```diff
@@ -0,0 +1,234 @@
+# MLFlow Tracking Server Operator
+
+## Overview
+
+In this repository we are providing our data scientists with tooling to perform hyper
+parameter tuning with  Kubernetes/OpenShift. Our business requirement is to have the
+ability to track parameters and metrics from their machine learning job
+in Kubernetes/OpenShift. We researched many tools but at this time the one that
+fits our use case for experiment tracking is MLFlow. We will be using
+the experiment tracking feature provided by MLFlow. We have ported
+MLFlow to run on OpenShift and is now available for data scientists to
+utilize.
+
+![](./imgs/image19.png) 
+The goal is for experiment tracking is to
+allow our data scientists to test many parameters at scale using
+OpenShift or Kubernetes. You can then compare
+different runs of your machine learning job with different parameters
+and choose the best parameters that yield the best metrics. See below
+screenshot.
+
+![](./imgs/image15.png) 
+
+## Preparing Images With S2I 
+ 
+To run your experiments you first need to get your code into docker
+images and install the python dependencies. We have provided a template
+and by exporting your jupyter notebook into a python file you can check
+it into your github. Make sure to set the APP\_FILE to the name of the
+file you exported. We have provided a template (mlflow-job-bc.yaml) to
+make that process simple to set up.
+
+![](./images/media/image9.png) 
+
+## Installing Tracking Server Operator
+ 
+At the moment we have decided to go with a shared MLFlow tracking server
+and each data scientist will have their own experiment sandbox (more on
+this later on in this document). In case you need to upgrade or want
+to install your own instance of tracking then the following command
+should allow for you to install mlflow tracking server in your namespace:
+
+Note: If your using minishift or minikube you must have cluster-admin priviledges to 
+install `ai_v1alpha1_trackingserver_crd.yaml`
+
+
+```bash
+# Setup Service Account
+$ kubectl create -f deploy/service_account.yaml
+# Setup RBAC
+$ kubectl create -f deploy/role.yaml
+$ kubectl create -f deploy/role_binding.yaml
+# Setup the CRD
+$ kubectl create -f deploy/crds/ai_v1alpha1_trackingserver_crd.yaml
+# Deploy the app-operator
+$ kubectl create -f deploy/operator.yaml
+```
+
+Note: Once the tracking server operator is installed then you can deploy the custom resource:
+ 
+ ```bash
+ # Create an TrackingServer CR
+ # The default controller will watch for TrackingServer objects and create a pod for each CR
+ $ kubectl create -f deploy/crds/ai_v1alpha1_trackingserver_cr.yaml
+ ```
+ 
+ In the `ai_v1alpha1_trackingserver_cr.yaml` file you can select which version of MLFlow you would like to run by
```

Comment:
What is the motivation behind this name, in examples `v1alpha1`?

Reply:
Just a naming for this is the first version and its alpha. But we can change that. In kubernetes you must have a version.  As we get newer versions of the operator we will change this version name. Similar to :

https://github.com/mlflow/mlflow/blob/0cfe8010599e5891df55dc102af4ee9675523d56/mlflow/version.py#L4

in mlflow

Reply:
But we can change that please advise what version you recommend that I put?

Reply:
For the example can we call `kubernetes_example`or something?